sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/CosmosBulkAsyncTest.java

```diff
@@ -840,6 +859,189 @@ private void createItemsAndVerify(List<com.azure.cosmos.models.CosmosItemOperati
+                    operationResponse.getResponse();
+
+                assertThat(operationResponse.getOperation()).isEqualTo(cosmosItemOperations.get(i));
+                if (i == 0) {
+                    assertThat(cosmosBulkItemResponse.getStatusCode()).isEqualTo(HttpResponseStatus.CREATED.code());
+                } else {
+                    assertThat(cosmosBulkItemResponse.getStatusCode()).isEqualTo(HttpResponseStatus.OK.code());
+                }
+                assertThat(cosmosBulkItemResponse.getRequestCharge()).isNotNull();
+                assertThat(cosmosBulkItemResponse.getRequestCharge()).isGreaterThan(0);
+                assertThat(cosmosBulkItemResponse.getCosmosDiagnostics().toString()).isNotNull();
+                assertThat(cosmosBulkItemResponse.getSessionToken()).isNotNull();
+                assertThat(cosmosBulkItemResponse.getActivityId()).isNotNull();
+
+            }
+        } finally {
+            if (executor != null && !executor.isDisposed()) {
+                executor.dispose();
+            }
+            if (rule != null) {
+                rule.disable();
+            }
+        }
+    }
+
+    // Tests No Retry Exception flow
+    @Test(groups = { "simple" }, timeOut = TIMEOUT)
+    public void executeBulk_preserveOrdering_OnServiceUnAvailable() {
+        int totalRequest = 100;
+
+        List<CosmosItemOperation> cosmosItemOperations = new ArrayList<>();
+        String duplicatePK = UUID.randomUUID().toString();
+        String duplicateId = UUID.randomUUID().toString();
+        PartitionKey duplicatePartitionKey = new PartitionKey(duplicatePK);
+        for (int i = 0; i < totalRequest; i++) {
+            if (i == 0) {
+                BatchTestBase.EventDoc eventDoc = new BatchTestBase.EventDoc(duplicateId, 2, 4, "type1",
+                    duplicatePK);
+                cosmosItemOperations.add(CosmosBulkOperations.getCreateItemOperation(eventDoc,
+                    duplicatePartitionKey));
+            } else {
+                cosmosItemOperations.add(CosmosBulkOperations.getPatchItemOperation(duplicateId,
+                    new PartitionKey(duplicatePK),
+                    CosmosPatchOperations.create().replace("/type", "updated" + i)));
+            }
+        }
+
+        CosmosBulkExecutionOptions cosmosBulkExecutionOptions = new CosmosBulkExecutionOptions();
+        ImplementationBridgeHelpers.CosmosBulkExecutionOptionsHelper
+            .getCosmosBulkExecutionOptionsAccessor()
+            .setOrderingPreserved(cosmosBulkExecutionOptions, true);
+
+        FaultInjectionRule rule = injectBatchFailure("ServiceUnavailable", FaultInjectionServerErrorType.SERVICE_UNAVAILABLE, 1);
+
+        Flux<CosmosItemOperation> inputFlux = Flux
+            .fromIterable(cosmosItemOperations)
+            .delayElements(Duration.ofMillis(100));
+        final BulkExecutorWithOrderingPreserved<BulkExecutorTest> executor = new BulkExecutorWithOrderingPreserved<>(
```

Comment:
Should this still be BulkExecutorTest? Shouldn't this be CosmosBulkAsyncTest?

Reply:
Changed to CosmosBulkAsyncTest